test/Npgsql.Tests/PrepareTests.cs

```diff
@@ -53,6 +58,43 @@ public async Task Async()
             }
         }
 
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/2925")]
+        public async Task Bug2925()
+        {
+            if (Type.GetType("Mono.Runtime") != null)
+                Assert.Ignore("Flaky on mono");
+
+            var builder = new NpgsqlConnectionStringBuilder(ConnectionString)
+            {
+                WriteBufferSize = NpgsqlWriteBuffer.MinimumSize,
+            };
+            await using var postmasterMock = new PgPostmasterMock(builder.ConnectionString);
+            using var _ = CreateTempPool(postmasterMock.ConnectionString, out var connectionString);
+            await using var conn = await OpenConnectionAsync(connectionString);
+            var server = postmasterMock.GetPendingServer();
+
+            var sb = new StringBuilder();
+            for (var i = 0; i < 1000; i++)
+            {
+                sb.Append($"SELECT {i};");
+            }
+            using var cmd = new NpgsqlCommand(sb.ToString(), conn);
+            // Async read wouldn't complete until async write from prepare is completed (on linux, on windows is ok)
+            // So, to make them run in parallel, we put it in another thread
+            var breakThread = new Thread(server.BreakOnRead);
```

Comment:
nit: if the intent is to invoke BreakOnRead from a separate thread, it's easier to just use Task.Run

Reply:
Oh, that is how it was at first. But interestingly enough, it doesn't work that way. 
As we're sending data to the localhost, it's super fast. And essentially, all of the write is synchronous and the break side doesn't get any quant of time.